### DISCUSSION TITLE
pulsed experiments

### DISCUSSION THREAD

**Discussion by Jingfu2000** [Ulm-IQO/qudi-iqo-modules#232] [General]
Dear qudi team,
now I come to pulsed experiment. I can run the experiments, such as rabi, but I did not get any signal. I now have the problems: 
![IMG_7979](https://github.com/user-attachments/assets/cceccb93-01c4-40c5-9e9b-da526efa91ec)

1)  I did not get counters in the fast counter, see the picture for the warning message. 
2) There seem  some parameters to be set. Where can I find the instructions?  

I also upload the config here.  thanks. 

#  Example Qudi configuration file.
#  The recommended path for user specific files is C:\Users\<user_name>\qudi\config

global:
    # list of modules to load when starting
    startup_modules: []

    # Module server configuration for accessing qudi GUI/logic/hardware modules from remote clients
    remote_modules_server:
        address: 'localhost'
        port: 12345

    # Server port for serving the active qudi module namespace locally (localhost).
    # Used by e.g. the Qudi jupyter kernel.
    namespace_server_port: 18861

    # If this flag is set (True), all arguments passed to qudi module APIs from remote
    # (jupyter notebook, qudi console, remote modules) will be wrapped and passed "per value"
    # (serialized and de-serialized). This is avoiding a lot of inconveniences with using numpy in
    # remote clients.
    # If you do not want to use this workaround and know what you are doing, you can disable this
    # feature by setting this flag to False.
    force_remote_calls_by_value: True

    # Qss stylesheet for controlling the appearance of the GUIs.
    # Absolute path or relative to qudi.artwork.styles
    stylesheet: 'qdark.qss'

    # Default root directory for measurement data storage. All eventual data sub-directories should
    # be contained within this directory. This is not enforced, just convention.
    # The fallback directory is <user home>/qudi/Data/
    # default_data_dir: C:\Users\neverhorst\qudi\Data

    # Save data to daily data sub-directories by default
    daily_data_dirs: True

hardware:

    mw_source_smiq:
        module.Class: 'microwave.mw_source_smiq.MicrowaveSmiq'
        options:
            visa_address: 'GPIB0::10::INSTR'
            comm_timeout: 20000  # in milliseconds

    mw_source_anritsu:
        module.Class: 'microwave.mw_source_anritsu.MicrowaveAnritsu'
        options:
            visa_address: 'USB0::0x0B5B::0xFFE0::112905::0::INSTR'
            comm_timeout: 10  # in seconds, optional

    fastcomtec_mcs6:
        module.Class: 'fastcomtec.fastcomtecmcs6.FastComtec'
        options:
            gated: False
            trigger_safety: 400e-9
            aom_delay: 390e-9
            minimal_binwidth: 1e-10

    nicard_6343_instreamer_confocal:
        module.Class: 'ni_x_series.ni_x_series_in_streamer.NIXSeriesInStreamer'
        options:
            device_name: 'Dev1'
            digital_sources:  # optional
                - 'PFI6'
            # external_sample_clock_source: 'PFI0'  # optional
            # external_sample_clock_frequency: 1000  # optional

    ni_finite_sampling_input_confocal:
        module.Class: 'ni_x_series.ni_x_series_finite_sampling_input.NIXSeriesFiniteSamplingInput'
        options:
            device_name: 'Dev1'
            digital_channel_units:  # optional
                'PFI6': 'c/s'
#            analog_channel_units:  # optional
            odmr_trigger_channel: '/Dev1/PFI7'   # JZ, Oct 08, 2024
#                'ai0': 'V'
#                'ai1': 'V'
            # external_sample_clock_source: 'PFI0'  # optional
            # external_sample_clock_frequency: 1000  # optional
#            adc_voltage_range: [-10, 10]  # optional, default [-10, 10]
#            max_channel_samples_buffer: 10000000  # optional, default 10000000
#            read_write_timeout: 10  # optional, default 10
#            sample_clock_output: '/Dev1/PFI20'  # optional, release by JZ, Oct 08, 24,
#            trigger_edge: RISING  # optional   release by JZ, Oct 08, 24

    ni_finite_sampling_io_confocal:
        module.Class: 'ni_x_series.ni_x_series_finite_sampling_io.NIXSeriesFiniteSamplingIO'
        options:
            device_name: 'Dev1'
            input_channel_units:
                PFI6: 'c/s'
            output_channel_units: # Specify used output channels
                'ao0': 'V'
                'ao1': 'V'
                'ao2': 'V'
                'ao3': 'V'
            adc_voltage_ranges:
                ai0: [-10, 10]  # optional
                ai1: [-10, 10]  # optional
            output_voltage_ranges:
                ao0: [0, 3]
                ao1: [0, 3]
                ao2: [0, 3]
                ao3: [0, 3]

    nicard_63XX_ao_confocal:
        module.Class: 'ni_x_series.ni_x_series_analog_output.NIXSeriesAnalogOutput'
        options:
            device_name: 'Dev1'
            channels:
                ao0:
                    limits: [0, 3]
                    keep_value: True
                ao1:
                    limits: [0, 3]
                    keep_value: True
                ao2:
                    limits: [0, 3]
                    keep_value: True
                ao3:
                    limits: [0, 3]
                    keep_value: True

    ni_scanning_probe_confocal:
        module.Class: 'interfuse.ni_scanning_probe_interfuse.NiScanningProbeInterfuse'
        # to use without tilt correction
        # module.Class: 'interfuse.ni_scanning_probe_interfuse.NiScanningProbeInterfuseBare'
        connect:
            scan_hardware: 'ni_finite_sampling_io_confocal'
            analog_output: 'nicard_63XX_ao_confocal'
        options:
            ni_channel_mapping:
                x: 'ao0'
                y: 'ao1'
                z: 'ao2'
                APD1: 'PFI6'
#                APD2: 'PFI9'
#                AI0: 'ai0'
            position_ranges: # in m
                x: [0, 30e-6]
                y: [0, 30e-6]
                z: [-2.15e-6, 2.15e-6]
            frequency_ranges: #Aka values written/retrieved per second; Check with connected HW for sensible constraints.
                x: [1, 5000]
                y: [1, 5000]
                z: [1, 1000]
            resolution_ranges:
                x: [1, 10000]
                y: [1, 10000]
                z: [2, 1000]
            input_channel_units:
                APD1: 'c/s'
#                APD2: 'c/s'
#                AI0: 'V'
            move_velocity: 400e-6 #m/s; This speed is used for scanner movements and avoids jumps from position to position.
            default_backward_resolution: 50

    pulser_awg70000:
        module.Class: 'awg.tektronix_awg70k.AWG70K'
        options:
            awg_visa_address: 'TCPIP0::192.168.0.87::inst0::INSTR'
            awg_ip_address: '192.168.0.87'
            timeout: 60
            # tmp_work_dir: 'C:\\Software\\qudi_pulsed_files' # optional
            # ftp_root_dir: 'C:\\inetpub\\ftproot' # optional, root directory on AWG device
            # ftp_login: 'anonymous' # optional, the username for ftp login
            # ftp_passwd: 'anonymous@' # optional, the password for ftp login

logic:

    time_series_reader_logic_confocal:
        module.Class: 'time_series_reader_logic.TimeSeriesReaderLogic'
        options:
            max_frame_rate: 20  # optional (default: 20Hz)
            channel_buffer_size: 1048576  # optional (default: 1MSample)
            max_raw_data_bytes: 1073741824  # optional (default: 1GB)
        connect:
            streamer: nicard_6343_instreamer_confocal

    scanning_probe_logic_confocal:
        module.Class: 'scanning_probe_logic.ScanningProbeLogic'
        options:
            max_history_length: 20
            max_scan_update_interval: 2
            position_update_interval: 1
        connect:
            scanner: ni_scanning_probe_confocal

    scanning_data_logic_confocal:
        module.Class: 'scanning_data_logic.ScanningDataLogic'
        options:
            max_history_length: 20
        connect:
            scan_logic: scanning_probe_logic_confocal

    scanning_optimize_logic_confocal:
        module.Class: 'scanning_optimize_logic.ScanningOptimizeLogic'
        connect:
            scan_logic: scanning_probe_logic_confocal

    poi_manager_logic_confocal:
        module.Class: 'poi_manager_logic.PoiManagerLogic'
        connect:
            scanning_logic: 'scanning_probe_logic_confocal'
            optimize_logic: 'scanning_optimize_logic_confocal'
            data_logic: 'scanning_data_logic_confocal'

    odmr_logic_confocal:
        module.Class: 'odmr_logic.OdmrLogic'
        connect:
            microwave: mw_source_anritsu
            data_scanner: ni_finite_sampling_input_confocal
        options:
            default_scan_mode: 'JUMP_LIST'  # optional

    pulsed_master_logic:
        module.Class: 'pulsed.pulsed_master_logic.PulsedMasterLogic'
        connect:
            pulsedmeasurementlogic: 'pulsed_measurement_logic'
            sequencegeneratorlogic: 'sequence_generator_logic'

    pulsed_measurement_logic:
        module.Class: 'pulsed.pulsed_measurement_logic.PulsedMeasurementLogic'
        options:
            raw_data_save_type: 'text'
        connect:
            fastcounter: 'fastcomtec_mcs6'
            pulsegenerator: 'pulser_awg70000'

    sequence_generator_logic:
        module.Class: 'pulsed.sequence_generator_logic.SequenceGeneratorLogic'
        options:
            assets_storage_path: C:\Users\CryoAtto\PycharmProjects\qudi_new_core\pulsed_assets
        connect:
            pulsegenerator: 'pulser_awg70000'

gui:

    time_series_gui_confocal:
        module.Class: 'time_series.time_series_gui.TimeSeriesGui'
        options:
            use_antialias: True  # optional, set to False if you encounter performance issues
        connect:
            _time_series_logic_con: time_series_reader_logic_confocal

    scanner_gui_confocal:
        module.Class: 'scanning.scannergui.ScannerGui'
        options:
            image_axes_padding: 0.02
            default_position_unit_prefix: null  # optional, use unit prefix characters, e.g. 'u' or 'n'
            optimizer_plot_dimensions: [2, 1]   # optimization sequence dim. order, here for 2D, 1D, e.g. XY, Z
            min_crosshair_size_fraction: 0.02   # minimum crosshair size as fraction of the displayed scan range
        connect:
            scanning_logic: scanning_probe_logic_confocal
            data_logic: scanning_data_logic_confocal
            optimize_logic: scanning_optimize_logic_confocal

    odmr_gui_confocal:
        module.Class: 'odmr.odmrgui.OdmrGui'
        connect:
            odmr_logic: 'odmr_logic_confocal'

    pulsed_gui:
        module.Class: 'pulsed.pulsed_maingui.PulsedMeasurementGui'
        connect:
            pulsedmasterlogic: 'pulsed_master_logic'

**Comment by Jingfu2000** [comment]
Dear all,
Today I checked the marker from AMG for generating the clock signal for the fast counter, and found that there is not output signal. This seems the problem for the zero photon detected by the fast counter.  
How should I solve this problem?  Thanks.

**Reply by ImPECcable012** [reply]
Hello,

I have the same problem right now, Did you manage to find a solution?

**Reply by Jingfu2000** [reply]
I solved this problem, but now I cannot recall the exact solution.  
Perhaps I reset something like this slide.   
![fast_counter](https://github.com/user-attachments/assets/1d21ed1c-76b2-428d-b1cc-8a371b7d83a6)
